src/qibolab/instruments/rfsoc.py

```diff
@@ -269,7 +268,7 @@ def _open_connection(host: str, port: int, server_commands: dict):
         # open a connection
         with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
             sock.connect((host, port))
-            msg_encoded = pickle.dumps(server_commands)
+            msg_encoded = bytes(json.dumps(server_commands), "utf-8")
```

Comment:
If sending `json` around becomes one of the bottlenecks, we can always consider using a more compact format (e.g. [`bson`](https://pymongo.readthedocs.io/en/stable/api/bson/)) or compressing the bytes before sending them.

But this is only relevant if you're frequently sending a lot of information.

Reply:
For what I saw the communication has very little impact on the total needed time so I'd say that this is not a problem